### PR TITLE
[Dependency] Have #merge take into account podspec repos

### PR DESCRIPTION
**Pull request by segiddins**

### CODE REVIEW

**Comment by dnkoutso** [comment]
missing `CHANGELOG` entry but 👍

File: spec/dependency_spec.rb [hunk 1]
```diff
@@ -224,6 +237,42 @@ module Pod
...
+      it 'allows merging if both dependencies have the same podspec repo' do
+        dep1 = Dependency.new('bananas', '~> 1.9', :source => 'https://source.git')
+        dep2 = Dependency.new('bananas', :source => 'https://source.git')
+
+        expected = Dependency.new('bananas', '~> 1.9', :source => 'https://source.git')
+        dep1.merge(dep2).should == expected
+        dep2.merge(dep1).should == expected
+      end
+
+      it 'raises when attempting to merge with a podspec repo and external source' do
```
Comment (dnkoutso) [inline spec/dependency_spec.rb]: is this the same test as above?

Comment (dnkoutso) [reply spec/dependency_spec.rb]: ah nevermind its external repo

File: lib/cocoapods-core/dependency.rb [hunk 2]
```diff
@@ -95,8 +95,7 @@ def initialize(name = nil, *requirements)
             raise Informative, 'A dependency with a specified podspec repo may ' \
               "not include other source parameters (#{name})."
           end
-        else
-          @external_source = additional_params
+        elsif @external_source = additional_params
           unless requirements.empty?
             raise Informative, 'A dependency with an external source may not ' \
```
Comment (justinseanmartin) [inline lib/cocoapods-core/dependency.rb]: I don't understand when/why we would expect dependencies with an external source to not specify a version requirement.

File: lib/cocoapods-core/dependency.rb [hunk 3]
```diff
@@ -249,24 +249,41 @@ def merge(other)
...
             end
 
+      opts = {}
+
       if external_source || other.external_source
-        self_external_source  = external_source || {}
-        other_external_source = other.external_source || {}
-        dep.external_source = self_external_source.merge(other_external_source)
+        opts.
+          merge!(external_source || {}) { raise ArguementError, 'key conflict' }.
```
Comment (justinseanmartin) [inline lib/cocoapods-core/dependency.rb]: This exception isn't helpful, should update to include which key conflicts and what their values are.

Comment (dnkoutso) [reply lib/cocoapods-core/dependency.rb]: Also `ArgumentError`

File: lib/cocoapods-core/dependency.rb (hunk 2 above)
Comment (segiddins) [reply lib/cocoapods-core/dependency.rb]: agreed, but I dont think this is the place to change that

**Comment by dnkoutso** [comment]
same needs a rebase

**Comment by segiddins** [comment]
Feedback addressed

**Pull request merged**